modern/src/settings/MaintenancePage.js

```diff
@@ -0,0 +1,161 @@
+    if (attribute && attribute.dataType) {
+      switch (attribute.dataType) {
+        case 'distance':
+          setLabels({ ...labels, start: t(prefixString('shared', distanceUnit)), period: t(prefixString('shared', distanceUnit))});
+          break;
+        case 'speed':
+          setLabels({ ...labels, start: t(prefixString('shared', speedUnit)), period: t(prefixString('shared', speedUnit))});
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  const rawToValue = (value) => {
+
+    const attribute = positionAttributes[item.type];
+    if (attribute && attribute.dataType) {
+      switch (attribute.dataType) {
+        case 'speed':
+          return speedFromKnots(value, speedUnit);
+        case 'distance':
+          return distanceFromMeters(value, distanceUnit);
+        default:
+          return value;
+      }
+    }
+
+    return value;
+  }
+
+  const valueToRaw = (value) => {
+
+    const attribute = positionAttributes[item.type];
+    if (attribute && attribute.dataType) {
+      switch (attribute.dataType) {
+        case 'speed':
+          return speedToKnots(value, speedUnit);
+        case 'distance':
+          return distanceToMeters(value, distanceUnit);
+        default:
+          return value;
+      }
+    }
+
+    return value;
+  }
+
+  return (
+    <EditItemView endpoint="maintenance" item={item} setItem={setItem}>
+      {item &&
+        <>
+          <Accordion defaultExpanded>
+            <AccordionSummary expandIcon={<ExpandMoreIcon />}>
+              <Typography variant="subtitle1">
+                {t('sharedRequired')}
+              </Typography>
+            </AccordionSummary>
+            <AccordionDetails className={classes.details}>
+              <TextField
+                margin="normal"
+                value={item.name || ''}
+                onChange={event => setItem({...item, name: event.target.value})}
+                label={t('sharedName')}
+                variant="filled" />
+              <FormControl variant="filled" margin="normal" fullWidth>
+                <InputLabel>{t('sharedType')}</InputLabel>
+                <Select 
+                  value={item.type || ''} 
+                  onChange={onMaintenanceTypeChange}>
+                    {convertToList(positionAttributes).map(({ key, name })=>(
+                      <MenuItem key={key} value={key}>{name}</MenuItem>
+                    ))}
+                </Select>
+              </FormControl>            
+              <TextField
+                margin="normal"
+                type="number"
+                value={rawToValue(item.start) || ''}
+                onChange={event => setItem({...item, start: valueToRaw(event.target.value)})}
+                label={t('maintenanceStart')}
+                variant="filled"
+                InputProps={{
+                endAdornment: <InputAdornment position="start">{labels.start}</InputAdornment>,
```

Comment:
What does `position` mean here?

Reply:
![image](https://user-images.githubusercontent.com/41992346/107914721-0b8e2600-6f89-11eb-9718-28982b6c1ef5.png)

Reply:
But isn't it at the end on the screenshot?

Reply:
This position attribute work in the top-down direction. For start and end, we use startAndornment and endAdornment.

![image](https://user-images.githubusercontent.com/41992346/108023081-d8f73280-7047-11eb-9dcf-5a18ce7f6af3.png)